tests/components/hyperion/test_light.py

```diff
@@ -0,0 +1,430 @@
+
+
+def create_mock_client():
+    """Create a mock Hyperion client."""
+    mock_client = Mock()
+    mock_client.async_client_connect = CoroutineMock(return_value=True)
+    mock_client.adjustment = None
+    mock_client.effects = None
+    mock_client.id = "%s:%i" % (TEST_HOST, TEST_PORT)
+    return mock_client
+
+
+def call_registered_callback(client, key, *args, **kwargs):
+    """Call a Hyperion entity callback that was registered with the client."""
+    return client.set_callbacks.call_args[0][0][key](*args, **kwargs)
+
+
+async def setup_entity(hass, client=None):
+    """Add a test Hyperion entity to hass."""
+    client = client or create_mock_client()
+    with patch("hyperion.client.HyperionClient", return_value=client):
+        assert await async_setup_component(
+            hass,
+            DOMAIN,
+            {
+                DOMAIN: {
+                    "platform": "hyperion",
+                    "name": TEST_NAME,
+                    "host": TEST_HOST,
+                    "port": const.DEFAULT_PORT,
+                    "priority": TEST_PRIORITY,
+                }
+            },
+        )
+    await hass.async_block_till_done()
+
+
+async def test_setup_platform(hass):
+    """Test setting up the platform."""
+    client = create_mock_client()
+    await setup_entity(hass, client=client)
+    assert hass.states.get(TEST_ENTITY_ID) is not None
+
+
+async def test_setup_platform_not_ready(hass):
+    """Test the platform not being ready."""
+    client = create_mock_client()
+    client.async_client_connect = CoroutineMock(return_value=False)
+
+    await setup_entity(hass, client=client)
+    assert hass.states.get(TEST_ENTITY_ID) is None
+
+
+async def test_light_basic_properies(hass):
+    """Test the basic properties."""
+    client = create_mock_client()
+    await setup_entity(hass, client=client)
+
+    entity_state = hass.states.get(TEST_ENTITY_ID)
+    assert entity_state.state == "on"
+    assert entity_state.attributes["brightness"] == 255
+    assert entity_state.attributes["hs_color"] == (0.0, 0.0)
+    assert entity_state.attributes["icon"] == hyperion_light.ICON_LIGHTBULB
+    assert entity_state.attributes["effect"] == hyperion_light.KEY_EFFECT_SOLID
+
+    # By default the effect list is the 3 external sources + 'Solid'.
+    assert len(entity_state.attributes["effect_list"]) == 4
+
+    assert (
+        entity_state.attributes["supported_features"] == hyperion_light.SUPPORT_HYPERION
+    )
+
+
+async def test_light_async_turn_on(hass):
+    """Test turning the light on."""
+    client = create_mock_client()
+    await setup_entity(hass, client=client)
+
+    # On (=), 100% (=), solid (=), [255,255,255] (=)
+    client.async_send_set_color = CoroutineMock(return_value=True)
+    await hass.services.async_call(
+        DOMAIN, SERVICE_TURN_ON, {ATTR_ENTITY_ID: TEST_ENTITY_ID}, blocking=True
+    )
+
+    assert client.async_send_set_color.call_args == call(
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6233107)